Source/SwiftLintFramework/Extensions/Dictionary+SwiftLint.swift

```diff
@@ -151,6 +151,50 @@ public struct SourceKittenDictionary {
     }
 }
 
+extension SourceKittenDictionary {
+    /// Traversing all substuctures of the dictionary hierarchically, calling |traverseBlock| on each node.
+    /// Traversing using depth first strategy, so deepest substructures will be passed to |traversBlock| first
+    /// - Parameter traverseBlock: block that will be called for each substructure in the dictionary
+    func traverseDepthFirst<T>(traverseBlock: (SourceKittenDictionary) -> [T]?) -> [T] {
+        var result: [T] = []
+        traverseDepthFirst(collectingValuesInto: &result, traverseBlock: traverseBlock)
+        return result
+    }
+
+    private func traverseDepthFirst<T>(collectingValuesInto array:inout [T],
+                                       traverseBlock: (SourceKittenDictionary) -> [T]?) {
+        substructure.forEach { subDict in
+            subDict.traverseDepthFirst(collectingValuesInto: &array, traverseBlock: traverseBlock)
+
+            if let collectedValues = traverseBlock(subDict) {
+                array += collectedValues
+            }
+        }
+    }
+
+    /// Traversing all substuctures of the dictionary hierarchically, calling |traverseBlock| on each node.
+    /// Traversing using depth first strategy, so deepest substructures will be passed to |traversBlock| first
+    /// - Parameter traverseBlock: block that will be called for each substructure and his parent
+    func traverseWithParentDepthFirst<T>(traverseBlock: (SourceKittenDictionary, SourceKittenDictionary) -> [T]?)
+        -> [T] {
+        var result: [T] = []
+        traverseWithParentDepthFirst(collectingValuesInto: &result, traverseBlock: traverseBlock)
+        return result
+    }
+
+    private func traverseWithParentDepthFirst<T>(
+        collectingValuesInto array:inout [T],
```

Comment:
Nit: space between `:` and `inout`:

```suggestion
        collectingValuesInto array: inout [T],
```